Tests/UnitTests/IdentityStateTests.swift

```diff
@@ -575,10 +594,207 @@ class IdentityStateTests: XCTestCase {
+                          source: EventSource.requestReset,
+                          data: nil)
+
+        let xdmSharedStateExpectation = XCTestExpectation(description: "XDM shared state should be updated once")
+        state.resetIdentifiers(event: event,
+                               createXDMSharedState: { _, _ in xdmSharedStateExpectation.fulfill() },
+                               eventDispatcher: { event in
+                                // Checking both the expected hit and that consent request event not sent
+                                XCTAssertTrue(event.type == EventType.edgeIdentity && event.source == EventSource.resetComplete)
+                                if event.type == EventType.edgeConsent && event.source == EventSource.updateConsent {
+                                    XCTFail("Consent request event should not be dispatched")
+                                }
+                               })
+
+        wait(for: [xdmSharedStateExpectation], timeout: 1)
+        XCTAssertFalse(mockDataStore.dict.isEmpty) // identity properties should have been saved to persistence
+        XCTAssertNil(state.identityProperties.advertisingIdentifier)
+    }
+
+    func testResetIdentitiesAdIdIsNilDoesNotDispatchConsentEvent() {
+        var props = IdentityProperties()
+        props.advertisingIdentifier = nil
+        props.ecid = ECID().ecidString
+
+        state = IdentityState(identityProperties: props)
+
+        let event = Event(name: "Test event",
+                          type: EventType.edgeIdentity,
+                          source: EventSource.requestReset,
+                          data: nil)
+
+        let xdmSharedStateExpectation = XCTestExpectation(description: "XDM shared state should be updated once")
+        state.resetIdentifiers(event: event,
+                               createXDMSharedState: { _, _ in xdmSharedStateExpectation.fulfill() },
+                               eventDispatcher: { event in
+                                // Checking both the expected hit and that consent request event not sent
+                                XCTAssertTrue(event.type == EventType.edgeIdentity && event.source == EventSource.resetComplete)
+                                if event.type == EventType.edgeConsent && event.source == EventSource.updateConsent {
+                                    XCTFail("Consent request event should not be dispatched")
+                                }
+                               })
+
+        wait(for: [xdmSharedStateExpectation], timeout: 1)
+        XCTAssertFalse(mockDataStore.dict.isEmpty) // identity properties should have been saved to persistence
+        XCTAssertNil(state.identityProperties.advertisingIdentifier)
+    }
+
+    // MARK: updateAdvertisingIdentifier(...)
+    /// Test ad ID is updated from nil to valid value on first call, and consent true is dispatched
+    func testUpdateAdvertisingIdentifierUpdatesNilWithValidId() {
```

Comment:
Renaming suggestion to increase readability: testUpdateAdvertisingIdentifier_whenNil_thenChangedToValid
can be applied to the other tests also